scripts/generate_oks_session

```diff
@@ -84,15 +89,70 @@ def generate(bootfile, datafile, session_name):
     vhostdals[h] = vhdal
 
   # ###########################################################################
-  # Create a run controller for the segment. We say it runs on the last
-  # specified virtual host for now since this is not defined in the json
+  # Create a run controller for the segment. 
   # ###########################################################################
+  fsm = controller['statefulnode']['fsm']
+  for iface in fsm['interfaces']:
+    ifname = 'user-providedirun-number'
+    iface = dal.FSMinterface('dummy-if', name=ifname, commands=[])
+    db.update_dal(iface)
+  pre_trans = []
+  for pt in fsm['pre_transitions']:
+    xtrans = fsm['pre_transitions'][pt]
+    ptdal = dal.FSMxTransition(pt,
+                               order=xtrans['order'],
+                               mandatory=xtrans['mandatory'])
+    db.update_dal(ptdal)
+    pre_trans.append(ptdal)
+  post_trans = []
+  for pt in fsm['post_transitions']:
+    xtrans = fsm['post_transitions'][pt]
+    ptdal = dal.FSMxTransition(pt,
+                               order=xtrans['order'],
+                               mandatory=xtrans['mandatory'])
+    db.update_dal(ptdal)
+    post_trans.append(ptdal)
+  transitions = []
+  for trans in fsm['transitions']:
+    trans = dal.FSMtransition(trans['trigger'],
+                              source=trans['source'],
+                              dest=trans['dest'])
+    db.update_dal(trans)
+    transitions.append(trans)
+  cmd_sequences = []
+  for seq in fsm['command_sequences']:
+    sequence = []
+    for cmd in fsm['command_sequences'][seq]:
+      cmd_dal = dal.FSMCommand(cmd['cmd'],
+                               cmd=cmd['cmd'],
+                               optional=cmd['optional'])
+      db.update_dal(cmd_dal)
+      sequence.append(cmd_dal)
+    seqdal = dal.FSMsequence(seq, sequence=sequence)
+    db.update_dal(seqdal)
+    cmd_sequences.append(seqdal)
+  fsm = dal.FSMconfiguration('fsmConf-1',
+                             states = fsm['states'],
+                             initial_state = fsm['initial_state'],
+                             transitions = transitions,
+                             pre_transitions = pre_trans,
+                             post_transitions = post_trans,
+                             command_sequences=cmd_sequences)
+  db.update_dal(fsm)
+  bcaster = dal.RCBroadcatser('bcaster',
+                              type=controller['broadcaster']['type'],
+                              address=controller['broadcaster']['kafka_address'],
+                              publish_timeout=controller['broadcaster']['publish_timeout'])
+  db.update_dal(bcaster)
   name = "my-controller"
   args = f"--conf-type OKS {datafile} ${{PORT}} {name}"
```

Comment:
Not strictly related to this PR, but this is not how it is done in the [drunc controller code](https://github.com/DUNE-DAQ/drunc/blob/develop/src/drunc/controller/configuration.py#L10C14-L10C22). Right now, we use URIs. So that would look like: `oks:///path/to/oks/file.xml`.

Reply:
Yes, I changed that this morning, haven't pushed my update yet but I changed it to `oksconfig:{cfg_path}` (cfg_path only starts with single '/') but only if you give an extra option `--oks_controller'. It now defaults to `file://{ctrlr_path}` (ctrlr_path is the abolute path to controller.jso). This allows drun to start a controller configured by JSON rather than OKS while we haven't written the OKS parsing for the controller.

Reply:
Very good.